src/sagemaker/multidatamodel.py

```diff
@@ -0,0 +1,211 @@
+           arguments supported by ``Model`` constructor
+
+        Args:
+            model_name (str): The model name.
+            model_data_prefix (str): The S3 prefix where all the models artifacts (.tar.gz)
+                in a Multi-Model endpoint are located
+            model (sagemaker.Model): The Model object that would define the
+                SageMaker model attributes like vpc_config, predictors, etc.
+                If this is present, the attributes for MultiDataModel are copied from this model
+                but are replaced the local arguments if present
+            image (str): A Docker image URI. If not provided, `model.image` is used instead.
+                (default: None)
+            role (str): An AWS IAM role (either name or full ARN). The Amazon
+                SageMaker training jobs and APIs that create Amazon SageMaker
+                endpoints use this role to access training data and model
+                artifacts. After the endpoint is created, the inference code
+                might use the IAM role if it needs to access some AWS resources.
+                It can be null if this is being used to create a Model to pass
+                to a ``PipelineModel`` which has its own Role field. If not provided,
+                `model.image` is used instead. (default: None)
+            predictor_cls (callable[string, sagemaker.session.Session]): A
+                function to call to create a predictor (default: None). If not
+                None, ``deploy`` will return the result of invoking this
+                function on the created endpoint name.
+            **kwargs: Keyword arguments passed to the ``Model`` initializer.
+        """
+        self.model_name = model_name
+        self._model_data_prefix = model_data_prefix
+        self.model_data_prefix = model_data_prefix
+        self.model = model
+        self.container_mode = MULTI_MODEL_CONTAINER_MODE
+        self.image = image
+        self.role = role
+        self.predictor_cls = predictor_cls
+
+        # Copy the values from the model if these attributes are not provided in the constructor
+        if self.model:
+            self.image = self.image or self.model.image
+            self.role = self.role or self.model.role
+            self.predictor_cls = self.predictor_cls or self.model.predictor_cls
+
+        super(MultiDataModel, self).__init__(
+            self.model_data_prefix,
+            self.image,
+            self.role,
+            name=self.model_name,
+            predictor_cls=self.predictor_cls,
+            **kwargs
+        )
+
+        if not self.sagemaker_session:
+            self.sagemaker_session = sagemaker.session.Session()
+        self.s3_client = self.sagemaker_session.boto_session.client("s3")
+
+        # Create the S3 prefix path to ensure model deployment succeeds
+        self._create_s3_model_data_path()
```

Comment:
Removed this as part of new commit. I had this in here to ensure that the create model does not fail in case `model_data_prefix` path does not exist yet (i.e. there are no models added to `MultiDataModel` object before `deploy()`. Makes sense to let the error propagate to the user if this occurs. Thanks for the suggestion.